File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 5:21
        w, r = 0, 0
        while r < len(nums):
            if nums[r] == nums[w]:
                r += 1
            else:
                w += 1
                nums[w] = nums[r]
                r += 1
        return w+1

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int w = 0;
        
        /**
        12345
          r
          w
        
        11122
            r
         w
        12
        */
        
        for (int r = 0; r < nums.length; r++) {
            // if (nums[r] == nums[w]) {
            //     continue();    
            // } else {
            //     w += 1;
            // }
            if (nums[r] != nums[w]) w += 1;
            nums[w] = nums[r];
        }
        return w+1;
            
    }
}